rlclientlib/live_model_impl.cc

```diff
@@ -142,10 +142,19 @@ namespace reinforcement_learning {
     }
   }
 
-  int live_model_impl::init_trace(api_status* status) {
-    const auto trace_impl = _configuration.get(name::TRACE_LOG_IMPLEMENTATION, value::NULL_TRACE_LOGGER);
+  int live_model_impl::init_trace(api_status* status, i_trace* binding_trace) {
+
     i_trace* plogger;
-    RETURN_IF_FAIL(_trace_factory->create(&plogger, trace_impl,_configuration, nullptr, status));
+
+    if (binding_trace != nullptr) {
+      const auto trace_impl = _configuration.get(name::TRACE_LOG_IMPLEMENTATION, value::BINDING_TRACE_LOGGER);
+      RETURN_IF_FAIL(_trace_factory->create(&plogger, trace_impl, _configuration, binding_trace, status));
```

Comment:
This is not working the way you think it's working. BINDING_TRACE_LOGGER is the default parameter in the case it's not defined in the configuration.

Reply:
We may misunderstand each other here - The type of TRACE_LOGGER is registered in factory_resolver.cc line 66. In the case, of undefined, we need to use the value/ default value to trigger the factory method.

---
In reply to: [252725392](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252725392) [](ancestors = 252725392)

Reply:
Fixed. The config is read from client.json. To enable this, we have to add "trace.logger.implementation": "BINDING_TRACE_LOGGER" in client.json file. 

After we refactor the configuration, it will be much easier to update.

---
In reply to: [252725392](https://github.com/VowpalWabbit/reinforcement_learning/pull/38#discussion_r252725392) [](ancestors = 252725392)